src/lib/device-types/index.ts

```diff
@@ -135,8 +154,7 @@ function updateDeviceTypesCache(
 	freshDeviceTypes: Promise<Dictionary<DeviceTypeInfo>>,
 ) {
 	if (!deviceTypesCache) {
-		deviceTypesCache = freshDeviceTypes;
-		return freshDeviceTypes;
+		return freshDeviceTypes.return();
```

Comment:
Removes the redundant `deviceTypesCache = freshDeviceTypes;` so that this only gets assigned from a within `fetchDeviceTypesAndReschedule`.
No need to return anything from in here.